### PR TITLE
Implemented integration status display in diagram

### PR DESCRIPTION
**Pull request by dngomez**
Now the diagram sources updates their state depending on `xxIntegration` boolean, gotten from the `guideState` subscription.

### CODE REVIEW

File: src/components/Panels/Guider/Diagram/Diagram.tsx [hunk 1]
```diff
@@ -225,8 +261,14 @@ function Flow() {
     // High-O
     let highoState: string;
     if (state.m1CorrectionsEnable) {
+      // Check if any source is active
+      const highoActive = sourceEdges.filter((n) => n.target === 'higho').some((n) => isSourceActive(n.id));
```
Comment (hugo-vrijswijk) [inline src/components/Panels/Guider/Diagram/Diagram.tsx:265]: Why not combine these into one? 
```suggestion
      const highoActive = sourceEdges.some((n) => n.target === 'higho' && isSourceActive(n.id));
```

File: src/components/Panels/Guider/Diagram/Diagram.tsx [hunk 2]
```diff
@@ -135,19 +152,20 @@ function Flow() {
...
-        animated: true,
-        className: 'active',
+        animated: isActive,
+        className: isActive ? 'active' : 'inactive',
       });
     }
 
     if (state.m1CorrectionsEnable) {
       const pos = sourceNodes.findIndex((n) => n.id === state.m2ComaM1CorrectionsSource);
+      const isActive = isSourceActive(state.m2ComaM1CorrectionsSource ?? '');
```
Comment (hugo-vrijswijk) [inline src/components/Panels/Guider/Diagram/Diagram.tsx:162]: Suggestion: have the `issourceActive` function accept `string | undefined` and then you won't need the `?? ''` anymore

Comment (dngomez) [reply src/components/Panels/Guider/Diagram/Diagram.tsx:162]: Will do!

File: src/components/Panels/Guider/Diagram/Diagram.tsx (hunk 1 above)
Comment (dngomez) [reply src/components/Panels/Guider/Diagram/Diagram.tsx:265]: No reason at all, just wasn't thinking straight.

**Review by hugo-vrijswijk — APPROVED** [review]

**Pull request merged**